packages/std/src/serde.rs

```diff
@@ -8,26 +8,75 @@ use serde::{de::DeserializeOwned, Serialize};
 use crate::binary::Binary;
 use crate::errors::{StdError, StdResult};
 
+#[deprecated = "use from_json_slice instead"]
 pub fn from_slice<T: DeserializeOwned>(value: &[u8]) -> StdResult<T> {
-    serde_json_wasm::from_slice(value).map_err(|e| StdError::parse_err(type_name::<T>(), e))
+    from_json_slice(value)
 }
 
+#[deprecated = "use from_json_binary instead"]
 pub fn from_binary<T: DeserializeOwned>(value: &Binary) -> StdResult<T> {
-    from_slice(value.as_slice())
+    from_json_binary(value)
+}
+
+/// Deserializes the given JSON byte vector to a data structure.
+///
+/// Errors if the input is not valid JSON or cannot be deserialized to the given type.
+pub fn from_json_slice<T: DeserializeOwned>(value: &[u8]) -> StdResult<T> {
+    serde_json_wasm::from_slice(value).map_err(|e| StdError::parse_err(type_name::<T>(), e))
+}
+
+/// Deserializes the given JSON bytes to a data structure.
+///
+/// Errors if the input is not valid JSON or cannot be deserialized to the given type.
+pub fn from_json_str<T: DeserializeOwned>(value: &str) -> StdResult<T> {
+    serde_json_wasm::from_str(value).map_err(|e| StdError::parse_err(type_name::<T>(), e))
+}
+
+/// Deserializes the given JSON bytes to a data structure.
+///
+/// Errors if the input is not valid JSON or cannot be deserialized to the given type.
+pub fn from_json_binary<T: DeserializeOwned>(value: &Binary) -> StdResult<T> {
+    from_json_slice(value.as_slice())
 }
 
+#[deprecated = "use to_json_vec instead"]
 pub fn to_vec<T>(data: &T) -> StdResult<Vec<u8>>
+where
+    T: Serialize + ?Sized,
+{
+    to_json_vec(data)
+}
+
+#[deprecated = "use to_json_binary instead"]
+pub fn to_binary_replace<T>(data: &T) -> StdResult<Binary>
```

Comment:
Here something went wrong with the function name

Reply:
oops, good catch